From 32a33fd268330026e79ea670f31647d038f2c224 Mon Sep 17 00:00:00 2001
From: Kevin Funk <kfunk@kde.org>
Date: Fri, 17 Apr 2020 11:52:54 +0200
Subject: [PATCH 1/1] Revert "Fix windows build"

That patch was wrong. BUILD_TESTING needs to always be defined when
testing is enabled so the *_TESTS_EXPORT macros are filled out properly.

I think this patch initially just worked around an issue which was fixed
by https://phabricator.kde.org/D20091.

Phabricator Revisino: https://phabricator.kde.org/D28759

This reverts commit 2f552ac573770ce22ec5489b22c6269280aa4e36.

Reviewers: vonreth, dvratil, mlaurent, dfaure, winterz

Reviewed By: mlaurent, dfaure

Subscribers: kfunk, vonreth, kde-pim

Tags: #kde_pim

Differential Revision: https://phabricator.kde.org/D28759
---
 CMakeLists.txt                      | 4 ++++
 src/akonadi-contacts/CMakeLists.txt | 4 +---
 src/contact-editor/CMakeLists.txt   | 4 +---
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0366341e..f315deeb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,6 +61,10 @@ endif()
 
 add_definitions(-DQT_NO_FOREACH)
 
+if(BUILD_TESTING)
+   add_definitions(-DBUILD_TESTING)
+endif()
+
 add_subdirectory(src)
 add_subdirectory(kcm)
 add_subdirectory(serializers)
diff --git a/src/akonadi-contacts/CMakeLists.txt b/src/akonadi-contacts/CMakeLists.txt
index 860ec195..7cca7893 100644
--- a/src/akonadi-contacts/CMakeLists.txt
+++ b/src/akonadi-contacts/CMakeLists.txt
@@ -113,9 +113,7 @@ ki18n_wrap_ui(akonadicontact_LIB_SRC contactgroupeditor.ui)
 
 
 add_library(KF5AkonadiContact ${akonadicontact_LIB_SRC})
-if (BUILD_TESTING)
-    target_compile_definitions(KF5AkonadiContact PRIVATE BUILD_TESTING)
-endif()
+
 generate_export_header(KF5AkonadiContact BASE_NAME akonadi-contact)
 
 add_library(KF5::AkonadiContact ALIAS KF5AkonadiContact)
diff --git a/src/contact-editor/CMakeLists.txt b/src/contact-editor/CMakeLists.txt
index 8fc2f8d2..16c607a7 100644
--- a/src/contact-editor/CMakeLists.txt
+++ b/src/contact-editor/CMakeLists.txt
@@ -142,9 +142,7 @@ set(contacteditor_LIB_SRC
 
 
 add_library(KF5ContactEditor ${contacteditor_LIB_SRC})
-if (BUILD_TESTING)
-    target_compile_definitions(KF5ContactEditor PRIVATE BUILD_TESTING)
-endif()
+
 generate_export_header(KF5ContactEditor BASE_NAME contacteditor)
 
 add_library(KF5::ContactEditor ALIAS KF5ContactEditor)
-- 
2.20.0.windows.1

